File: strings/Common child/common_child_hari.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the commonChild function below.
def commonChild(s1, s2):
    x = len(s1)
    y = len(s2)
    n = [[0 for _ in range(y+1)] for _ in range(x+1)]
    for i in range(1,x+1):
        for j in range(1,y+1):
            if s1[i-1]==s2[j-1]:
                n[i][j]=1+n[i-1][j-1]
            else:
                n[i][j] =max(n[i-1][j],n[i][j-1])
    return n[x][y] 



if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s1 = input()

    s2 = input()

    result = commonChild(s1, s2)

    fptr.write(str(result) + '\n')

    fptr.close()

File: Hackerrank/strings/Common child/Common_child_hari.cpp
#include <bits/stdc++.h>

using namespace std;
static int T[5001][5001];

// Complete the commonChild function below.
int commonChild(string s1, string s2) {
    int m = s1.size();
    int n = s2.size();
    int i,j,k,kk;
    for(i = 0; i <= m; i++ ){
        for(j = 0; j <= n; j++ ){
            if(i == 0 || j == 0){
                T[i][j] = 0;
            }else if(s1[i-1]==s2[j-1]){
                T[i][j] = 1+T[i-1][j-1];
            }else{
                T[i][j] = max(T[i][j-1], T[i-1][j]);
            }
        }
        //cout<<i<<" "<<j<<"\n";
        //for(k=0;k<=m;k++)
        //{
        //    for(kk=0;kk<=n;kk++)
        //   {
        //        cout<<T[k][kk]<<" ";
        //    }
        //    cout<<"\n";
        //}
    }
    return(T[m][n]);
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s1;
    getline(cin, s1);

    string s2;
    getline(cin, s2);

    int result = commonChild(s1, s2);

    fout << result << "\n";

    fout.close();

    return 0;
}
